### PR TITLE
Update App.Navigation.md - function components

### PR DESCRIPTION
**Pull request by ygreking**
Update navigation section with function components code snippets

### CODE REVIEW

File: docs/App.Navigation.md [hunk 1]
```diff
@@ -201,6 +239,45 @@ class PostsList extends Component {
...
+<summary>With Functional Components</summary>
+
+```js
+import React, {Component} from 'react';
+import {View, Text, StyleSheet} from 'react-native';
+import {Navigation} from 'react-native-navigation';
+
+const PostsList = (props) => {
+
+  const pushViewPostScreen = () => {
```
Comment (bogekt) [inline docs/App.Navigation.md:253]: @ygreking Please, wrap each all callback with useCallback hook

File: docs/App.Navigation.md [hunk 2]
```diff
@@ -388,6 +554,62 @@ class AddPost extends Component {
...
+<summary>With Functional Components</summary>
+
+```js
+...
+import {View, Text, TextInput, StyleSheet} from 'react-native';
+...
+const AddPost = (props) => {
+  ...
+
+  const onChangeText = text => {
```
Comment (bogekt) [inline docs/App.Navigation.md:568]: @ygreking Please, wrap each all callback with useCallback hook

File: docs/App.Navigation.md [hunk 3]
```diff
@@ -388,6 +554,62 @@ class AddPost extends Component {
...
+        rightButtons: [{
+          id: 'saveBtn',
+          text: 'Save',
+          enabled: !!text
+        }]
+      }
+    });
+  }
+
+  render() {
```
Comment (bogekt) [inline docs/App.Navigation.md:580]: remove render, just return needed as its a FC

**Review by bogekt — CHANGES_REQUESTED** [review]

File: docs/App.Navigation.md (hunk 1 above)
Comment (ygreking) [reply docs/App.Navigation.md]: done

File: docs/App.Navigation.md (hunk 2 above)
Comment (ygreking) [reply docs/App.Navigation.md]: done

**Review by bogekt — APPROVED** [review]

File: docs/App.Navigation.md [hunk 4]
```diff
@@ -416,7 +639,7 @@ By this point you have:
 * Learned how to [style the navigator](https://wix.github.io/react-native-navigation/docs/style-options) using the Options object and how to dynamically merge options.
 
 You can view the full project in [this repository](
-https://github.com/wix-playground/wix-mobile-crash-course).
+https://github.com/wix-playground/wix-mobile-crash-course). It's originally written with class components. Functional components version can be found in [this branch](https://github.com/wix-incubator/react-native-crash-course/tree/functional-components-hooks)
```
Comment (bogekt) [inline docs/App.Navigation.md:642]: do we plan to merge this or left for all the time?

Comment (ygreking) [reply docs/App.Navigation.md:642]: let's merge and update docs accordingly

**Review by bogekt — APPROVED** [review]

**Pull request merged**